PivotCalibration/Logic/vtkSlicerPivotCalibrationLogic.cxx

```diff
@@ -427,3 +404,80 @@ void vtkSlicerPivotCalibrationLogic::GetToolTipToToolMatrix( vtkMatrix4x4* matri
 {
   matrix->DeepCopy( this->ToolTipToToolMatrix );
 }
+
+//---------------------------------------------------------------------------
+void vtkSlicerPivotCalibrationLogic::SetToolTipToToolMatrix( vtkMatrix4x4* matrix )
+{
+  this->ToolTipToToolMatrix->DeepCopy( matrix );
+}
+
+//---------------------------------------------------------------------------
+vnl_vector< double > vtkSlicerPivotCalibrationLogic::ComputeSecondaryAxis( vnl_vector< double > shaftAxis_ToolTip )
+{
+  // If the secondary axis 1 is parallel to the shaft axis in the tooltip frame, then use secondary axis 2
+  vnl_vector< double > orthogonalAxis_Shaft( 3, 3, ORTHOGONAL_AXIS );
+  double angle = acos( dot_product( shaftAxis_ToolTip, orthogonalAxis_Shaft ) );
+  // Force angle to be between -pi/2 and +pi/2
+  if ( angle > vtkMath::Pi() / 2 )
+  {
+    angle -= vtkMath::Pi();
+  }
+  if ( angle < - vtkMath::Pi() / 2 )
+  {
+    angle += vtkMath::Pi();
+  }
+
+  if ( fabs( angle ) < vtkMath::RadiansFromDegrees( PARALLEL_ANGLE_THRESHOLD_DEGREES ) ) // If shaft axis and orthogonal axis are not parallel
+  {
+    return vnl_vector< double >( 3, 3, BACKUP_AXIS );
+  }
+  return orthogonalAxis_Shaft;
+}
+
+
+//---------------------------------------------------------------------------
+void vtkSlicerPivotCalibrationLogic::VerifyShaftDirection()
```

Comment:
This method doesn't just verify the direction and returns a status but actually changes the direction. So, it should be rename to something like UpdateShaftDirection.